src/components/views/login/PasswordLogin.js

```diff
@@ -63,14 +75,29 @@ module.exports = React.createClass({displayName: 'PasswordLogin',
 
     onSubmitForm: function(ev) {
         ev.preventDefault();
-        this.props.onSubmit(this.state.username, this.state.password);
+        this.props.onSubmit(
+            this.state.username,
+            this.state.phoneCountry,
+            this.state.phoneNumber,
+            this.state.password,
+        );
     },
 
     onUsernameChanged: function(ev) {
         this.setState({username: ev.target.value});
         this.props.onUsernameChanged(ev.target.value);
     },
 
+    onPhoneCountryChanged: function(country) {
+        this.setState({phoneCountry: country});
```

Comment:
again, why are we tracking this (and `phoneNumber`) as a separate bit of state, rather than passing it in as a prop? it violates the keep-your-state-in-one-place principle of react.